### PR TITLE
Patch inspection

### PR DESCRIPTION
**Pull request by jason-rodri**
Determining if these code segments need to added to out fips offering

### CODE REVIEW

File: random/random-drbg.c [hunk 1]
```diff
@@ -334,6 +334,9 @@ enum drbg_prefixes
  * Global variables
```
Comment (jason-rodri) [inline random/random-drbg.c:334]: https://github.com/gpg/libgcrypt/commit/f436bf4451cb2ad0a1f56aa843bf79a58878022f

File: tests/t-kdf.c [hunk 2]
```diff
@@ -1497,6 +1497,63 @@ check_argon2 (void)
     goto again;
 }
```
Comment (jason-rodri) [inline tests/t-kdf.c:1498]: https://lists.gnutls.org/pipermail/gcrypt-devel/2022-July/005346.html

File: random/random-drbg.c (hunk 1 above)
Comment (jason-rodri) [reply random/random-drbg.c:334]: https://dev.gnupg.org/T5933

File: cipher/rsa.c [hunk 3]
```diff
@@ -183,14 +183,76 @@ test_keys_fips (gcry_sexp_t skey)
...
+  gcry_sexp_t decr  = NULL;
+  char *decr_plaintext = NULL;
+  int decr_plaintext_len = 0, strip;
+  gcry_sexp_t tmplist = NULL;  
+  gcry_mpi_t ref_mpi = NULL;
+  int counter;
 
-  /* Create a random plaintext.  */
-  _gcry_randomize (plaintext, sizeof plaintext, GCRY_WEAK_RANDOM);
+    /* Create a random plaintext.  */
```
Comment (jason-rodri) [inline cipher/rsa.c]: @jallisonciq This update to test_keys_fips doesn't look right, but I think it was added to support the rsa PTC enc/dec feature

Comment (jason-rodri) [reply cipher/rsa.c]: looks like the code was taken from https://github.com/ciq-rocky-fips/libgcrypt/blob/9b90cfd7f1a94cf79103f5ce787297856c617042/cipher/rsa.c#L2196

File: tests/t-kdf.c (hunk 2 above)
Comment (jallisonciq) [reply tests/t-kdf.c:1498]: This looks correct.

File: cipher/rsa.c (hunk 3 above)
Comment (jallisonciq) [reply cipher/rsa.c]: Yes, this makes sense as encrypting random data and then checking it decrypts to the same value is useless for FIPS-140-3. An enc/dec implementation that merely copied plaintext would pass :-).

You have to encrypt a known plaintext with a known key and look for a known ciphertext to make sense.

FYI, just squash this patch: "cleaning up incorrectly patched code" into the previous patch. There's no sense in adding code just to remove it again in the next commit.

**Comment by jallisonciq** [comment]
Comments inline above. Firstly, can you squash the two patches together (There's no sense in adding code just to remove it again in the next commit). Secondly, can you add a *much* more comprehensive commit message, explaining what this is doing and adding the links to the upstream commits (or the commit refspecs and specific branch if it's a stable upstream git branch).

That way we have a good history of what was done and why.

With those changes LGTM !

**Pull request closed without merging**